File: Tests/WebApiController/ClientApp/src/app/edge-cases/models/self-referencing-model.ts
/* eslint-disable */
// tslint:disable

export class SelfReferencingModel {
    public name?: string;
    public children?: SelfReferencingModel[];

    public constructor(init?: Partial<SelfReferencingModel>) {
        Object.assign(this, init);
    }
}

// outputid:627408ca-a818-4326-b843-415f5bbfb028

File: Tests/AspDotNetAnnotations/Models/SelfReferencingModel.cs
using System.Collections.Generic;

namespace AspDotNetAnnotations.Models
{
    public class SelfReferencingModel
    {
        public string Name { get; set; } = "";
        public List<SelfReferencingModel> Children { get; } = new List<SelfReferencingModel>();
    }
}
